File: fapi/src/main/java/models/Meeting.java
package models;

import lombok.Data;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Data
public class Meeting {

    private long id;
    private String title;
    private Date start;
    private Date end;
    private User owner;
    private Room room;
    private List<User> members = new ArrayList<>();
}

File: front/src/app/modules/meeting/models/meeting.ts
import {CalendarEvent, CalendarEventAction} from 'angular-calendar';
import {Room} from '../../room/models/room';
import {User} from '../../user/models/user';

export interface EventColor {
  primary: string;
  secondary: string;
}

export class Meeting implements CalendarEvent {
  idMeeting: string;
  start: Date;
  title: string;
  end: Date;
  room: Room;
  owner: User;
  actions: CalendarEventAction[] = [];
  color: EventColor;

  constructor(idMeeting: string, title: string, start: Date, end: Date, room: Room, owner: User, actions: CalendarEventAction[],
              color?: EventColor) {
    this.idMeeting = idMeeting;
    this.title = title;
    this.start = new Date(start);
    this.end = new Date(end);
    this.room = room;
    this.owner = owner;
    this.actions = actions;
    this.color = color;
  }
}
